src/Ranger.php

```diff
@@ -234,14 +234,14 @@ private function prepare_date($input)
         if ($input instanceof DateTime) {
             return $input;
         }
-        if (is_string($input)) {
-            return new Datetime($input);
-        }
-        if (is_int($input)) {
+        if (is_numeric($input)) {
             $date = new Datetime;
             $date->setTimestamp($input);
```

Comment:
according to https://www.php.net/manual/en/datetime.settimestamp.php, `setTimestamp` expects an int as the parameter, so maybe it would be good to cast here (because otherwise I guess you would at least get notices or somthing if this gets called with a float)

Reply:
For example:

```php
declare(strict_types=1);
echo (new DateTime())->setTimestamp(1222222223.3222222);
```

Output:

```
Fatal error: Uncaught TypeError: DateTime::setTimestamp() expects parameter 1 to be int, float given in test.php:2 
Stack trace: 
#0 test.php(2): DateTime->setTimestamp() 
#1 {main} thrown in test.php on line 2
```

Reply:
alternatively, we could say `if (is_numeric($input) && !is_float($input)) {` above

Reply:
> alternatively, we could say `if (is_numeric($input) && !is_float($input)) {` above

I won't cast out floating numbers: although it might be rare (and useless) in this context, a timestamp can include microseconds, so instead of excluding them, I would rather use setTimestamp(intval($input)) to make sure an integer is passed, so it would meet any use case 

(actually I made tests it with floating numbers, and it worked fine, that's why I didn't include intval() in the fix, but this might be specific to my setup, I guess it's better to include it)